docs/guides/input.md

```diff
@@ -0,0 +1,292 @@
+# Input
+
+_Since version 4.27.0_
+
+[:rocket: Quick Reference](#quick-reference)
+
+When you're building software, it's common to split your code up into small pieces. These might be functions in a node process, or framework components. These small pieces often need to declare **dependencies** - bits of information/code that they're not responsible for, but that they rely on to serve their purpose.
+
+The best way to declare dependencies to an XState machine is called `input`. You can specify input like so:
+
+```ts
+const machine = createMachine(
+  {
+    entry: (context, event, meta) => {
+      console.log(meta.state.input.lyric); // 'In west Philadelphia, born and raised'
+    }
+  },
+  {
+    input: {
+      lyric: 'In west Philadelphia, born and raised'
+    }
+  }
+);
+```
+
+Or, by using the `.withConfig` method:
+
+```ts
+const machineWithDifferentInput = machine.withConfig({
+  input: {
+    lyric: 'In the playground, where I spend most of my days'
+  }
+});
+```
+
+The input is then available in any action/service/guard, in the `meta` argument.
+
+```ts
+const machine = createMachine(
+  {
+    entry: [
+      (context, event, meta) => {
+        console.log(meta.state.input.foo);
+      }
+    ],
+    invoke: {
+      src: (context, event, meta) => {
+        console.log(meta.state.input.foo);
+      },
+      onDone: {
+        cond: (context, event, meta) => {
+          console.log(meta.state.input.foo);
+          return false;
+        }
+      }
+    }
+  },
+  {
+    input: {
+      foo: 'bar'
+    }
+  }
+);
+```
+
+Unlike [context](./context.md), inputs cannot be assigned to from _inside_ the machine. If you do need to mutate the input locally, consider putting it in `context` instead.
+
+### Inputs that change over time
+
+You can change an input on a service that's running by calling `service.input()`.
+
+```ts
+const service = interpret(machine).start();
+
+service.input({
+  foo: 'bar'
+});
+```
+
+Under the hood, this sends a `xstate.input` event which XState recognises internally and uses to update the input.
+
+### Listening for changes
+
+You can listen for changes by listening for the `xstate.input` event:
+
+```ts
+const machine = createMachine({
+  on: {
+    'xstate.input': {
+      actions: (context, event, meta) => {
+        console.log(meta.state.input);
+      }
+    }
+  }
+});
+```
+
+Note that `meta` will be updated _before_ the actions and guards that listen for this transition are executed. This means you don't need to read the updated input from the event:
+
+```ts {5-6,8-9}
```

Comment:
what do those curly braces mean here? it's the first time I see this and I can't find any reference for this in https://github.github.com/gfm/#fenced-code-blocks

Reply:
They're for highlighting lines of code, used elsewhere in the repo

Reply:
Yep, they're for Vuepress